### PR TITLE
ellips updated reader + mock function

### PR DESCRIPTION
**Pull request by cmmngr**
Changes in ellips reader:

- Optional use of mock function for ellipsometry data
- Automatic labelling of data names depending on data type and angle of incidence
- Writing atom_types as list (if entered as string in eln_data.yaml)

### CODE REVIEW

File: nexusutils/dataconverter/readers/ellips/reader_utils.py [hunk 1]
```diff
@@ -0,0 +1,889 @@
...
+                    = self.restructure_value(element_.text,
+                                             element_.tag)
+        elif child_num == 1 \
+                and 'any' == element_.tag:
+            child_elmt = element_[0]
+            self._xps_dict[f'{parent_path}'] \
+                = self.restructure_value(child_elmt.text,
+                                         child_elmt.tag)
+
+    def check_for_siblins_with_same_name(self, reslv_name, new_sblings_elmt):
```
Comment (sanbrock) [inline nexusutils/dataconverter/readers/ellips/reader_utils.py:373]: typo

File: nexusutils/dataconverter/readers/ellips/reader_utils.py [hunk 2]
```diff
@@ -0,0 +1,889 @@
...
+        while child_num > 0:
+
+            self.pass_child_through_parsers(element_,
+                                            parent_path,
+                                            child_elmt_ind,
+                                            skip_child=-1)
+            child_num -= 1
+            child_elmt_ind += 1
+
+    def struct_fc_name_sc_value(self,
```
Comment (sanbrock) [inline nexusutils/dataconverter/readers/ellips/reader_utils.py:207]: fs and  sc are too expressive

File: nexusutils/dataconverter/readers/ellips/reader_utils.py [hunk 3]
```diff
@@ -0,0 +1,889 @@
...
+
+import xml.etree.ElementTree as EmtT
+from typing import Tuple, List, Any, Union
+import copy
+import xarray as xr
+import numpy as np
+from ase.data import chemical_symbols
+
+
+def extract_atom_types(formula: str) -> list:
```
[the remark this answers is not in the document]
Comment (cmmngr) [reply nexusutils/dataconverter/readers/ellips/reader_utils.py:30]: Rubel mentioned that 'extract_atom_types' now is in 'pynxtools/dataconverter/helper.py', so 'reader_utils' can be removed + path in mock.py needs to be updated

[the remark this answers is not in the document]
Comment (sanbrock) [reply nexusutils/dataconverter/readers/ellips/reader_utils.py]: yes, please merge master to this branch, so nexusutils is renamed into pynxtools already here.

**Pull request closed without merging**